plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java

```diff
@@ -102,13 +129,53 @@ public DockerConnectorConfiguration(URI dockerDaemonUri,
         this.networkFinder = networkFinder;
     }
 
-    public static String getExpectedLocalHost() {
-        return SystemInfo.isLinux() ? "localhost" : dockerDaemonUri().getHost();
+    public static boolean isUnixSocketUri(URI uri) {
+        return UNIX_SOCKET_SCHEME.equals(uri.getScheme());
+    }
+
+    public String getDockerHost() {
+        if (isUnixSocketUri(dockerDaemonUri)) {
+            return "localhost";
+        } else {
+            return getDockerHostIp();
+        }
+    }
+
+    /**
+     * Gets the Docker host ip address. This is host that can be reached from a docker container.
+     *
+     * @return docker host IP address
+     * @see <a href="https://docs.docker.com/articles/networking/">Docker Networking</a>
+     */
+    public String getDockerHostIp() {
+        return getDockerHostIp(SystemInfo.isLinux(), System.getenv());
+    }
+
+    public URI getDockerDaemonUri() {
+        return dockerDaemonUri;
+    }
+
+    public InitialAuthConfig getAuthConfigs() {
+        return authConfigs;
+    }
+
+    public DockerCertificates getDockerCertificates() {
+        if (dockerCertificatesDirectoryPath == null || !getDockerDaemonUri().getScheme().equals("https")) {
+            return null;
+        }
+        final File dockerCertificatesDirectory = new File(dockerCertificatesDirectoryPath);
+        return dockerCertificatesDirectory.isDirectory() ? DockerCertificates.loadFromDirectory(dockerCertificatesDirectory) : null;
+    }
+
+    /**
+     * For testing purposes
```

Comment:
Then it shouldn't be public, or maybe you forgot to remove either comment or method?